x-pack/solutions/observability/plugins/observability_onboarding/ui_tests/tests/custom_logs/configuration.spec.ts

```diff
@@ -0,0 +1,257 @@
+  });
+
+  test('should be optional allowing user to continue if it is empty', async ({
+    pageObjects: { customLogsPage },
+  }) => {
+    await customLogsPage.clickAdvancedSettingsButton();
+    await customLogsPage.logFilePathInput(0).fill('myLogs.log');
+    await expect(customLogsPage.customConfigInput()).toHaveValue('');
+    await expect(customLogsPage.continueButton()).not.toBeDisabled();
+  });
+
+  test.describe('Integration name', () => {
+    test.beforeEach(async ({ pageObjects: { customLogsPage } }) => {
+      await customLogsPage.logFilePathInput(0).fill('myLogs.log');
+    });
+
+    test('Users should not be able to continue if they do not specify an integrationName', async ({
+      pageObjects: { customLogsPage },
+    }) => {
+      await customLogsPage.integrationNameInput().fill('');
+      await expect(customLogsPage.continueButton()).toBeDisabled();
+    });
+
+    test('value will contain _ instead of special chars', async ({
+      pageObjects: { customLogsPage },
+    }) => {
+      await customLogsPage.integrationNameInput().fill('hello$world');
+      await expect(customLogsPage.integrationNameInput()).toHaveValue('hello_world');
+    });
+
+    test('value will be invalid if it is not lowercase', async ({
+      pageObjects: { customLogsPage },
+      page,
+    }) => {
+      await customLogsPage.integrationNameInput().fill('H3llowOrld');
+      await expect(page.getByText('An integration name should be lowercase.')).toBeVisible();
+    });
+  });
+
+  test.describe('datasetName', () => {
+    test.beforeEach(async ({ pageObjects: { customLogsPage } }) => {
+      await customLogsPage.logFilePathInput(0).fill('myLogs.log');
+    });
+
+    test('Users should not be able to continue if they do not specify a datasetName', async ({
+      pageObjects: { customLogsPage },
+    }) => {
+      await customLogsPage.datasetNameInput().fill('');
+      await expect(customLogsPage.continueButton()).toBeDisabled();
+    });
+
+    test('value will contain _ instead of special chars', async ({
+      pageObjects: { customLogsPage },
+    }) => {
+      await customLogsPage.datasetNameInput().fill('hello$world');
+      await expect(customLogsPage.datasetNameInput()).toHaveValue('hello_world');
+    });
+
+    test('value will be invalid if it is not lowercase', async ({
+      pageObjects: { customLogsPage },
+      page,
+    }) => {
+      await customLogsPage.datasetNameInput().fill('H3llowOrld');
+      await expect(page.getByText('A dataset name should be lowercase.')).toBeVisible();
+    });
+  });
+
+  test.describe('Custom integration', () => {
+    const CUSTOM_INTEGRATION_NAME = 'mylogs';
+
+    test.beforeEach(async ({ pageObjects: { customLogsPage } }) => {
+      await customLogsPage.deleteIntegration(CUSTOM_INTEGRATION_NAME);
+    });
+
+    test.describe('when user is missing privileges', () => {
+      test.beforeEach(async ({ browserAuth, page }) => {
```

Comment:
nit: `beforeAll` and `afterAll` here seem more suitable compared to `beforeEach` and `afterEach`.

Reply:
Agree, but the thing it, Playwright does not allow using `page` and `context` (which `browserAuth` uses) fixtures in `beforeAll`/`afterAll`. We'd need to create both fixtures manually, which does not seem to worth it in this case. [Playwrihte doc for reference](https://aka.ms/playwright/reuse-page), just in case.